auth/bitbucket/server/auth_config.go

```diff
@@ -0,0 +1,68 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"github.com/concourse/atc"
+	"github.com/concourse/atc/auth/bitbucket"
+	"github.com/concourse/atc/auth/routes"
+	"github.com/hashicorp/go-multierror"
+	"github.com/tedsuo/rata"
+)
+
+type AuthConfig struct {
+	ConsumerKey string           `json:"consumer_key" long:"consumer-key" description:"Application consumer key for enabling Bitbucket OAuth"`
+	PrivateKey  privateKeyConfig `json:"private_key" long:"private-key" description:"Application private key for enabling Bitbucket OAuth, in base64 encoded DER format"`
```

Comment:
This should probably be a path to a file instead, similar to [other flags](https://github.com/concourse/atc/blob/750c42a630d8fc35dc58053993a53e0cbe8e2de9/atccmd/command.go#L104) of the same nature.

Reply:
Wouldn't this require the transfer of the file to the machine running ATC though?

Reply:
Depends on the automation tooling you've got. Our binaries already expect files for more complicated values like private keys. Our BOSH release just dumps templates into files on the machine, and our Docker instructions pass it all along via a volume mount.

I just think it's cleaner than having the key show up in the command's argument list; that could be revealed by `ps` and such (or `/proc/XXX/cmdline`). Granted, files on a disk aren't the greatest either, but at least you can have permissions configured appropriately.

Reply:
I was under the impression that teams could modify their own auth, which would make putting the keys on the server problematic. But even if they could, the key would probably only have to be readable by `fly`, right?

Reply:
Yeah, it would only need to be readable by `fly`.